setup.py

```diff
@@ -191,5 +193,99 @@ def run_system_command(command: str):
         raise RuntimeError(f"Failed to execute command: '{command}'")
 
 
+def is_cuda_version_ok(cfg):
+    nvcc_process = subprocess.run("nvcc --version", shell=True, stdout=subprocess.PIPE)
+    nvcc_ver_match = re.search("V[0-9]+.[0-9]+.[0-9]+", nvcc_process.stdout.decode("utf-8"))
+    if not nvcc_ver_match:
+        raise RuntimeError("CUDA not found")
+    major = int(nvcc_ver_match[0].split(".")[0][1:])  # [1:] to remove char 'v'
+    minor = int(nvcc_ver_match[0].split(".")[1])
+    patch = int(nvcc_ver_match[0].split(".")[2])
+    print(f"Found CUDA {major}.{minor}.{patch}")
+    for (actual, expected) in [(major, cfg.CUDA_MIN_VER_MAJOR), (minor, cfg.CUDA_MIN_VER_MINOR),
+                               (patch, cfg.CUDA_MIN_VER_PATCH)]:
+        if actual > expected:
+            return True
+        if actual < expected:
+            return False
+    return True
+
+
+def install_pcl_deps(cfg):
+    # Clone vcpkg
+    if not os.path.isdir(cfg.VCPKG_DIR):
+        if on_linux() and not inside_docker():  # Inside docker already installed
+            print("Installing dependencies for vcpkg...")
+            run_system_command("sudo apt update")
+            run_system_command("sudo apt install git curl zip unzip tar freeglut3-dev libglew-dev libglfw3-dev")
+        run_subprocess_command(
+            f"git clone -b {cfg.VCPKG_TAG} --single-branch --depth 1 https://github.com/microsoft/vcpkg {cfg.VCPKG_DIR}")
+    # Bootstrap vcpkg
+    if not os.path.isfile(os.path.join(cfg.VCPKG_DIR, cfg.VCPKG_EXEC)):
+        run_subprocess_command(f"{os.path.join(cfg.VCPKG_DIR, cfg.VCPKG_BOOTSTRAP)}")
+
+    # Install dependencies via vcpkg
+    run_subprocess_command(
+        f"{os.path.join(cfg.VCPKG_DIR, cfg.VCPKG_EXEC)} install --clean-after-build pcl[core,visualization]:{cfg.VCPKG_TRIPLET}")
+
+
+def has_colcon():
+    process = subprocess.Popen("colcon --help", shell=True, stderr=subprocess.DEVNULL, stdout=subprocess.DEVNULL)
+    process.wait()
+    return process.returncode == 0
+
+
+def install_ros2_deps(cfg):
+    # Install colcon if needed
+    if not has_colcon():
+        if on_windows():
+            run_system_command("pip install colcon-common-extensions")
+        else:
+            run_system_command("sudo apt update")
+            run_system_command("sudo apt install python3-colcon-common-extensions")
+    # Clone radar msgs
+    if not os.path.isdir(cfg.RADAR_MSGS_DIR):
+        run_subprocess_command(
+            f"git clone --single-branch --depth 1 https://github.com/ros-perception/radar_msgs.git {cfg.RADAR_MSGS_DIR}")
+        run_subprocess_command(f"cd {cfg.RADAR_MSGS_DIR} && git checkout {cfg.RADAR_MSGS_COMMIT} && cd ..")
+    # Build radar msgs
+    if not os.path.isdir(cfg.RADAR_MSGS_INSTALL_DIR):
+        original_path = Path.cwd()
+        os.chdir(cfg.RADAR_MSGS_DIR)
+        run_subprocess_command(f"colcon build")
```

Comment:
The command fails if ROS2 is not sourced. I'd add a check (the same as `has_colcon()`) and inform the user to source ROS2 before if he didn't do it.

Reply:
Seems to not be addressed yet.